armi/reactor/reactors.py

```diff
@@ -2316,3 +2324,41 @@ def processLoading(self, cs, dbLoad: bool = False):
         self.p.maxAssemNum = self.getMaxParam("assemNum")
 
         getPluginManagerOrFail().hook.onProcessCoreLoading(core=self, cs=cs)
+
+    def _axiallyExpandAssems(
+        self, assems: list, dbLoad: bool, expTag: str, detAxExp: bool
+    ):
+        """expand assembliues"""
+        axialExpChngr = AxialExpansionChanger(detAxExp)
+        runLog.header(
+            f"=========== Axially expanding {expTag} assemblies (except control) from Tinput to Thot ==========="
+        )
+        for a in assems:
+            if not a.hasFlags(Flags.CONTROL):
+                axialExpChngr.setAssembly(a)
+                axialExpChngr.expansionData.computeThermalExpansionFactors()
+                axialExpChngr.axiallyExpandAssembly(thermal=True)
+        if not dbLoad:
+            axialExpChngr.manageCoreMesh(self.parent)
+        self._updateBlockBOLHeights(assems, dbLoad)
+
+    def _updateBlockBOLHeights(self, assems: list, dbLoad: bool):
+        """post thermal expansion, update block BOL heights
+
+        Parameters
+        ----------
+        assems: list
+            a list of :py:class:`Assembly <armi.reactor.assemblies.Assembly>` objects
+            that have been axially expanded
+        dbLoad: bool
+            used to determine if Core::processLoading is originating from database load or not
+        """
+        if not dbLoad:
+            self.p.referenceBlockAxialMesh = self.findAllAxialMeshPoints(
+                applySubMesh=False
+            )
+            self.p.axialMesh = self.findAllAxialMeshPoints()
+        for a in assems:
+            if not a.hasFlags(Flags.CONTROL):
+                for b in a:
+                    b.p.heightBOL = b.getHeight()
```

Comment:
I dont think this is going to work because you haven't called setBlock mesh on the blueprints yet, so they are not yet the final heights

Reply:
Good catch. 
https://github.com/terrapower/armi/pull/823/commits/d0ebfe51170939f45c11f3024492cabaf2eebab0